.github/workflows/r_package.yml

```diff
@@ -62,54 +62,54 @@ jobs:
           - os: ubuntu-latest
             task: r-package
             compiler: gcc
-            r_version: 4.3
+            r_version: latest
```

Comment:
For now, proposing that we use keys "latest" and "oldest" in the config and use the following specific versions in scripts:

| OS | oldest R | latest R |
|:---:|:--------:|:--------:|
| linux | 4.2.0 | 4.5.1 |
| macos | 4.0.5 | 4.5.1 |
| Windows | 4.3.1 | 4.5.1 |

It's difficult to test R 4.0.x on Linux in CI here because 4.2.0 is the oldest version available at https://cloud.r-project.org/bin/linux/ubuntu/jammy-cran40/

It's difficult to test R 4.0.x on Windows because `rtools40` bundled different tools than later RTools, and I hadn't yet been able to figure out how to get them to interact nicely with Visual Studio 2022: https://github.com/microsoft/LightGBM/actions/runs/17572546785